tests/bundle.test.js

```diff
@@ -0,0 +1,54 @@
+import test from "ava";
+import tokamak from "../src/tokamak.js";
+import { conclude } from "conclure";
+import { allSettled } from "conclure/combinators";
+
+test("integration test", async (t) => {
+  let requireGraph = {};
+
+  let fileMap = {
+    "file:///file1.js": 'import test from "ava";\nexport default 42;',
+  };
+
+  const loader = {
+    load: function* load(url) {
+      if (!url.startsWith("file://")) {
+        throw new Error(`Don't know how to load ${url}`);
+      }
+      // this would normally be async
+      requireGraph[url] = fileMap[url];
+      return {
+        id: url,
+        code: fileMap[url],
+      };
```

Comment:
It would be even more interesting if you keep the whole node `{ id, code }` in requireGraph. Then the `code` field will be transpiled by Tokamak, and `imports` and `required` fields added to the node, so you would be able to see the results for the whole tree after `yield allSettled`